tests/functional/spec/routing_and_skipping_section_dependencies.spec.js

```diff
@@ -0,0 +1,236 @@
+    $(SkipAgePage.submit()).click();
+    $(SkipSectionSummaryPage.submit()).click();
+
+    selectPrimaryPerson();
+    answerAndSubmitNameQuestion();
+    answerAndSubmitAgeQuestion();
+
+    $(ReasonNoConfirmationPage.iDidButItWasRemovedFromThePathAsIChangedMyAnswerToNoOnTheSkipQuestion()).click();
+    $(ReasonNoConfirmationPage.submit()).click();
+
+    expectPersonalDetailsName();
+    expectPersonalDetailsAge();
+    expect($(PrimaryPersonSummaryPage.reasonNoConfirmationAnswer()).getText()).to.contain(
+      "I did, but it was removed from the path as I changed my answer to No on the skip question"
+    );
+  });
+
+  it("When I answer 'Yes' to skipping the age question and populate the household, Then in each repeating section I am not asked my age", () => {
+    answerNoToSkipAgeQuestion();
+
+    addHouseholdMembers();
+
+    $(HubPage.summaryRowLink("household-personal-details-section-1")).click();
+    $(RepeatingSexPage.female()).click();
+    $(RepeatingSexPage.submit()).click();
+    $(RepeatingAgePage.answer()).setValue("45");
+    $(RepeatingAgePage.submit()).click();
+
+    expect($(HouseHoldPersonalDetailsSectionSummaryPage.repeatingSexAnswer()).getText()).to.contain("Female");
+    expect($(HouseHoldPersonalDetailsSectionSummaryPage.repeatingAgeAnswer()).getText()).to.contain("45");
+
+    $(HouseHoldPersonalDetailsSectionSummaryPage.submit()).click();
+    $(HubPage.summaryRowLink("household-personal-details-section-2")).click();
+    $(RepeatingSexPage.male()).click();
+    $(RepeatingSexPage.submit()).click();
+    $(RepeatingAgePage.answer()).setValue("10");
+    $(RepeatingAgePage.submit()).click();
+
+    expect($(HouseHoldPersonalDetailsSectionSummaryPage.repeatingSexAnswer()).getText()).to.contain("Male");
+    expect($(HouseHoldPersonalDetailsSectionSummaryPage.repeatingAgeAnswer()).getText()).to.contain("10");
+  });
+
+  it("When I answer 'No' to skipping the age question and populate the household, Then in each repeating section I am not asked my age", () => {
+    answerYesToSkipAgeQuestion();
+
+    addHouseholdMembers();
+
+    $(HubPage.summaryRowLink("household-personal-details-section-1")).click();
+    $(RepeatingSexPage.female()).click();
+    $(RepeatingSexPage.submit()).click();
+    expect($(HouseHoldPersonalDetailsSectionSummaryPage.repeatingSexAnswer()).getText()).to.contain("Female");
+    expect($(HouseHoldPersonalDetailsSectionSummaryPage.repeatingAgeAnswer()).isExisting()).to.be.false;
+
+    $(HouseHoldPersonalDetailsSectionSummaryPage.submit()).click();
+    $(HubPage.summaryRowLink("household-personal-details-section-2")).click();
+    $(RepeatingSexPage.male()).click();
+    $(RepeatingAgePage.submit()).click();
+
+    expect($(HouseHoldPersonalDetailsSectionSummaryPage.repeatingSexAnswer()).getText()).to.contain("Male");
+    expect($(HouseHoldPersonalDetailsSectionSummaryPage.repeatingAgeAnswer()).isExisting()).to.be.false;
+  });
+
+  function selectPrimaryPerson() {
```

Comment:
Minor, but we should look to use arrow functions when specific features of full functions are not needed.